exec/java-exec/src/main/java/org/apache/drill/exec/store/log/LogFormatPlugin.java

```
@@ -0,0 +1,84 @@
+import org.apache.drill.exec.server.DrillbitContext;
+import org.apache.drill.exec.store.RecordReader;
+import org.apache.drill.exec.store.RecordWriter;
+import org.apache.drill.exec.store.dfs.DrillFileSystem;
+import org.apache.drill.exec.store.dfs.easy.EasyFormatPlugin;
+import org.apache.drill.exec.store.dfs.easy.EasyWriter;
+import org.apache.drill.exec.store.dfs.easy.FileWork;
+import org.apache.hadoop.conf.Configuration;
+
+import java.io.IOException;
+import java.util.List;
+
+public class LogFormatPlugin extends EasyFormatPlugin<LogFormatConfig> {
+
+  public static final String DEFAULT_NAME = "log";
+  private final LogFormatConfig formatConfig;
+
+  public LogFormatPlugin(String name, DrillbitContext context,
+                         Configuration fsConf, StoragePluginConfig storageConfig,
+                         LogFormatConfig formatConfig) {
+    super(name, context, fsConf, storageConfig, formatConfig,
+        true,  // readable
+        false, // writable
+        true, // blockSplittable
+        true,  // compressible
+        Lists.newArrayList(formatConfig.extension),
+        DEFAULT_NAME);
+    this.formatConfig = formatConfig;
+  }
+
+  @Override
+  public RecordReader getRecordReader(FragmentContext context,
+                                      DrillFileSystem dfs, FileWork fileWork, List<SchemaPath> columns,
+                                      String userName) throws ExecutionSetupException {
+    return new LogRecordReader(context, dfs, fileWork,
+        columns, userName, formatConfig);
+  }
+
+  @Override
+  public boolean supportsPushDown() {
+    return true;
+  }
+
+  @Override
+  public RecordWriter getRecordWriter(FragmentContext context,
+                                      EasyWriter writer) throws IOException {
+    return null;
+  }
+
+  @Override
+  public int getReaderOperatorType() {
+    return 0;
+  }
+
+  @Override
+  public int getWriterOperatorType() {
+    return 0;
```

Please throw `UnsupportedOperationException` here.

---

Thrown